utils/firestore.js

```diff
@@ -780,32 +780,186 @@ const retrieveSiteNotifications = async (siteId, isParent) => {
+                if (key === 'token') queryInput = collection.where('token', '==', queries[key]);
+                if (key === 'studyId') queryInput = collection.where('state.studyId', '==', queries[key]);
+            }
+
+            return queryInput;
+        }        
+        
+  
+        // This function executes each query and pushes the data to the fetchedResults array
+        const executeQuery = async (query) => {
+            const snapshot = await (queries['allSiteSearch'] === 'true' ? query.get() : query.where('827220437', operator, siteCode).get());
+            if (snapshot.size !== 0) {
+                snapshot.docs.forEach(doc => {
+                    fetchedResults.push(doc.data());
+                });
+            }
+        };
+
+        // This function removes duplicate Connect_ID entries from fetchedResults. Since we have to perform multiple get() requests to handle the array-contains queries,
+        // We expect to end up with duplicate entries in fetchedResults that need to be handled.
+        // Example: A search for 'John Smith' will result in two get() requests: one for 'John' and one for 'Smith'. Both requests will return the same participant, so we need to remove the duplicate.
+        const removeDuplicateResults = () => {
+            let uniques = new Set();
+
+            fetchedResults = fetchedResults.filter(participant => {
+                if (uniques.has(participant.Connect_ID)) {
+                    return false;
+                } else {
+                    uniques.add(participant.Connect_ID);
+                    return true;
+                }
+            });
+        };
+
+        // Map from query type, returns true if a participant doc should be kept in fetchedResults
+        const queryToResultsChecker = {
+            firstName: (participant, queryValue) => {
+                const value = participant.query.firstName;
+                return typeof value === 'string'
+                    ? value === queryValue
+                    : Array.isArray(value) && value.includes(queryValue);
+            },
+            lastName: (participant, queryValue) => {
+                const value = participant.query.lastName;
+                return typeof value === 'string'
+                    ? value === queryValue
+                    : Array.isArray(value) && value.includes(queryValue);
+            },
+            email: (participant, queryValue) => participant.query.allEmails.includes(queryValue),
+            phone: (participant, queryValue) => participant.query.allPhoneNo.includes(queryValue),
+            dob: (participant, queryValue) => participant['371067537'] === queryValue,
+        };
+
+        // This function removes results that don't match the query. For example, if the query is for 'John Smith' and the results include 'John Doe', 'John Smith', and 'Jane Smith', we need to remove 'John Doe' and 'Jane Smith' from the results.
```

Comment:
I agree these descriptive notes are helpful but I suggest you condense them 1-2 lines or just remove it entirely.